onnx_xla/onnx_xla_interface.cc

```diff
@@ -267,3 +277,79 @@ ONNXIFI_SYMBOL_NAME(onnxReleaseGraph)(onnxGraph graph) {
   }
   ONNXIFI_CATCH_EXCPETION();
 }
+
+//Initialize event by creating EventControl object on the heap
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
+    onnxInitEvent)(onnxBackend backend, onnxEvent* event) {
+  try {
+    if (!backend)  {
+      return ONNXIFI_STATUS_INVALID_BACKEND;
+    }
+    if (!event)  {
+      return ONNXIFI_STATUS_INVALID_POINTER;
+    }
+
+    *event = (onnxEvent) new EventControl();
+    return ONNXIFI_STATUS_SUCCESS;
+  }
+  ONNXIFI_CATCH_EXCPETION();
+
+
+}
+
+//Signal Event by changing the signalled boolean under mutex hold
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
+    onnxSignalEvent)(onnxEvent event)  {
+  try {
+    auto *e = reinterpret_cast<EventControl *>(event);
+    if (!event)  {
+      return ONNXIFI_STATUS_INVALID_EVENT;
+    }
+    {
+       std::lock_guard<std::mutex> lk(e->m);
+       if (e->signalled)  {
+         return ONNXIFI_STATUS_INVALID_STATE;
+       }
+       e->signalled = true;
+    }
+    e->cv.notify_all();
+    return ONNXIFI_STATUS_SUCCESS;
+  }
+  ONNXIFI_CATCH_EXCPETION();
+}
+
+//Wait for signalled to be turned true using conditional variable to coordinate
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
+    onnxWaitEvent)(onnxEvent event)  {
+  try {
+    auto *e = reinterpret_cast<EventControl *>(event);
+    if (!event)  {
+      return ONNXIFI_STATUS_INVALID_EVENT;
+    }
+  
+    std::unique_lock<std::mutex> lk(e->m);
+    e->cv.wait(lk, [&e]{return e->signalled;});
+    lk.unlock();
+    e->cv.notify_all();
```

Comment:
You don't need to call this, because this function doesn't change `e->signalled`

Reply:
My thinking here was suppose multiple threads are waiting on the signal (A and B), then we need to wake them all up. After the first wakeup (A), the mutex is acquired by that thread (A). so once A is done with the mutex it should signal B to wake up so B can proceed as well.

Reply:
It should be the place that changes the status of the signal to notify all listeners.

Reply:
ok i do this, so deleting